external-crates/move/crates/move-cli/src/base/test.rs

```diff
@@ -179,8 +184,18 @@ pub fn run_move_unit_tests<CW: Write + Send, TW: Write + Send>(
     // Move package system, to first grab the compilation env, construct the test plan from it, and
     // then save it, before resuming the rest of the compilation and returning the results and
     // control back to the Move package system.
-    build_plan.compile_with_driver(compiler_writer, |compiler| {
-        let (files, comments_and_compiler_res) = compiler.run::<PASS_CFGIR>().unwrap();
+    let mut warning_diags = None;
+    build_plan.compile_with_driver(writer, |compiler| {
+        let (files, comments_and_compiler_res) = if no_lint || !sui_flavor {
+            compiler.run::<PASS_CFGIR>().unwrap()
+        } else {
+            let (filter_attr_name, filters) = known_filters();
+            compiler
+                .add_visitors(linter_visitors())
+                .add_custom_known_filters(filters, filter_attr_name)
```

Comment:
I wonder if this is the wrong place to add this. I feel like it would be better to do this inside of the package system so it would be used everywhere automagically. Thoughts?

Reply:
Good point! I wasn't sure how to get the right flags inside `CompiledPackage::build_all` but after looking a bit more closely I realized they are available in the resolution graph.